redash/utils/__init__.py

```diff
@@ -42,6 +42,10 @@ def dt_from_timestamp(timestamp, tz_aware=True):
 def slugify(s):
     return re.sub('[^a-z0-9_\-]+', '-', s.lower())
 
+def gen_signature_hash(parameters, key):
+    if key is None:
+        key = settings.EMBED_KEY
+    return hashlib.sha256(str(parameters)+key).hexdigest()
```

Comment:
You better sort the parameters, to make sure the order doesn't make a difference. Also look into using the [`hmac`](https://docs.python.org/2/library/hmac.html) library.